middleware.go

```diff
@@ -62,10 +63,10 @@ func (m *Middleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next htt
 	url := "other"
 	for _, u := range m.routes {
 		if strings.HasPrefix(r.URL.Path, u) {
-			url = u
+			url = r.URL.Path
```

Comment:
Hmmm is this ok?. This will record the complete path which might cause some issues with metric cardinality. 

I.E: If you have 3 requests to the following paths:

```
/users/1234
/users/5678
/users/9131
```
This will record three different metrics. I believe what we want to record with is just 3 requests to the `/users`  endpoint.

Reply:
Okay. I do not like the idea of manually putting the routes that generate metrics but at the moment it can work as it was before.

Reply:
Agree but I can't think of a better way of handling this as that information is usually stored in the multiplexer. Having one distinct metric per request just doesn't scale (https://prometheus.io/docs/practices/naming/#labels)